src/Core/Framework/Store/Services/StoreAppLifecycleService.php

```diff
@@ -67,11 +67,12 @@ public function removeExtensionAndCancelSubscription(int $licenseId, string $tec
         $app = $this->getAppById($id, $context);
         $this->storeClient->cancelSubscription($licenseId, $context);
         $this->appLifecycle->delete($technicalName, ['id' => $id, 'roleId' => $app->getAclRoleId()], $context);
-        $this->deleteExtension($technicalName);
+        $this->deleteExtension($technicalName, $context);
     }
 
-    public function deleteExtension(string $technicalName): void
+    public function deleteExtension(string $technicalName, Context $context): void
     {
+        $this->uninstallExtension($technicalName, $context, true);
```

Comment:
I need some feedback here:

I added this line because the extension isn't deleted from the database - meaning, even though the files are deleted, the extension is still in the database. Because the modal does not have the `Keep all user data` option, I opted to just keep user data by default.

Do you think this is sufficient? Or do you think it's better for me to add the radio button to the modal, and pass the value down and handle it correctly here?

For more clarity, this is the difference between the two modals:

Cancel and Remove
![image](https://github.com/user-attachments/assets/cc0b4807-4365-468d-aeae-3ab295b3a1c8)
Uninstall
![image](https://github.com/user-attachments/assets/b7ad594b-2c1b-44b2-b4b6-ac098d7c36c1)

Reply:
I would prefer the later, preferably without the huge margin